engine/src/main/java/org/terasology/world/generation/facets/base/BaseBooleanFieldFacet2D.java

```diff
@@ -31,8 +31,7 @@
 
     public BaseBooleanFieldFacet2D(BlockRegion targetRegion, Border3D border) {
         super(targetRegion, border);
-        Vector2i size = getRelativeRegion().size();
-        data = new boolean[size.x * size.y];
+        data = new boolean[getRelativeRegion().area()];
```

Comment:
That's way better, but as commented on the interface we should rename the method:
```suggestion
        data = new boolean[getRelativeArea().area()];
```